### PR TITLE
feat/td-32240 code review and quality improvement

### PR DESCRIPTION
**Pull request by YamingPei**
1. Remove the bridge function for taos_stmt_reclaim_fields, which is already provided on all platforms;
2. Add the parsing of valbinary/geometry data in WebSocket and add corresponding test cases;
3. SQLBindCol interface supports binary type.
4. Troubleshoot resource leaks in tag fields.

### CODE REVIEW

File: tests/c/c_test.c [hunk 1]
```diff
@@ -174,6 +174,75 @@ static int handles_init(handles_t *handles, const char *connstr)
   return -1;
 }
 
+static SQLSMALLINT sqltype_2_ctype(SQLLEN v, SQLLEN is_unsigned)
```
Comment (DuanKuanJun) [inline tests/c/c_test.c:177]: why the geometry type not found on the list ?

Comment (YamingPei) [reply tests/c/c_test.c:177]: This function is a conversion of the SQL type to the SQL_C type, which is defined in sql.h/sqlext.h by standard ODBC and has no GEOMETRY type

Comment (DuanKuanJun) [reply tests/c/c_test.c:177]: OK

**Review by DuanKuanJun — APPROVED** [review]

**Review by freemine — APPROVED** [review]
LGTM

**Review by sheyanjie-qq — APPROVED** [review]

**Pull request merged**